sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -1445,7 +1445,10 @@ object HiveExternalCatalog {
     case _: AnsiIntervalType => false
     case _: TimestampNTZType => false
     case _: VariantType => false
-    case s: StructType => s.forall(f => isHiveCompatibleDataType(f.dataType))
+    case s: StructType =>
```

Comment:
question: so for top-level columns, HMS accepts any kind of characters?

Reply:
Yes, the reason why nested struct columns are special is because HMS thrift API passes it as a plain string in the TYPE field of FieldSchema, the HMS server has to deserialize it from string to the Hive Struct. That's where it fails. Top-level names are put in the NAME field of FieldSchema.